views/widget.scout.admin.php

```diff
@@ -56,22 +47,33 @@
 				}
 			}
 		} catch ( Exception $e ) {
-			print 'We could not retrieve the SPM sections.';
+			write_log($e);
+			$spm_err = true;
 		}
 
 		?>
 
-		<?php if ( sailthru_spm_ready() ) : ?>
+		<?php if ( sailthru_spm_ready() && ! isset( $spm_err) ) : ?>
+
 			<input type="hidden" value="personalize_js" name="sailthru_widget_type" />
 			<div id="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>_div" style="display: block; margin:15px 0">
 				<p>Choose the personalization section to display on your site by selecting the section from the drop down menu below. </p>
 				<input type="hidden" value="personalize_js" name="sailthru_widget_type" />
-				<?php echo $section_dropdown; ?>
-				<p class="small">Manage Site this personalization block in <a href="https://my.sailthru.com/spm">Sailthru</a></p>
+				<?php /* there can be a case where SPM has been disabled - do a double check */ ?>
+				<?php if ( isset( $section_dropdown ) ): ?>
+					<?php echo $section_dropdown; ?>
```

Comment:
Missing escaping on `$section_dropdown`

Reply:
hmm or rather early escaping, eitherway this could be turned into a function that outputs internally, that takes the respone of the `apiGet` call as its parameter, HTML is near free to generate so no need to cache it